src/scanner/btc.go

```diff
@@ -213,7 +216,11 @@ func (s *BTCScanner) waitForNextBlock(block *CommonBlock) (*CommonBlock, error)
 	for {
 		nextBlock, err := s.getNextBlock(block)
 		if err != nil {
-			log.WithError(err).Error("getNextBlock failed")
+			if err.Error() == ErrEmptyBlock.Error() {
```

Comment:
compare `if err == ErrEmptyBlock`

Reply:
no, error can not compare directly , only string can compre

Reply:
why

Reply:
https://stackoverflow.com/questions/39121172/how-to-compare-golang-error-objects